### PR TITLE
Enabling Trusted Access for Mass Registration Script

### PR DESCRIPTION
**Pull request by jaskisin**
Enabling Trusted Access for Mass Registration Script

### CODE REVIEW

File: MassRegistrationOfSAPSystems/RegisterSIDs.ps1 [hunk 1]
```diff
@@ -67,38 +67,42 @@ foreach($line in $file)
...
-        $SubTag = $TagVal.Split("=")
-        $Keys += $SubTag[0]
-        $Values += $SubTag[1]
+        $command += " -ManagedResourceGroupStorageAccountName $ManagedRgStorageAccountName"
     }
-    $i=0
-    $Tag = @{}
-    foreach($Key in $Keys)
+
+    if ($ManagedResourcesNetworkAccessType -match '[a-zA-Z]' -or $ManagedResourcesNetworkAccessType -match '[0-9]')
```
Comment (ajaygit158) [inline MassRegistrationOfSAPSystems/RegisterSIDs.ps1:87]: Can we match it to Public and Private as the only values supported if supplied.

Comment (jaskisin) [reply MassRegistrationOfSAPSystems/RegisterSIDs.ps1]: This is done

**Review by ajaygit158 — APPROVED** [review]

File: MassRegistrationOfSAPSystems/RegistrationData.csv [hunk 2]
```diff
@@ -1,2 +1,2 @@
-SID,Location,Environment,Product,CentralServerVmId,MsiId,ManagedResourceGroupName,ManagedRgStorageAccountName,Tag,State
-TST,eastus,NonProd,S4HANA,/subscriptions/<subscription>/resourceGroups/massreg/providers/Microsoft.Compute/virtualMachines/tstvm,/subscriptions/<subscription>/resourcegroups/JASKIRAT-RG/providers/Microsoft.ManagedIdentity/userAssignedIdentities/jaskirat-msi,mrg-TST,acssstoragetst,key1 = value1; key2 = value2; key3 = value3; key4 = value4,
+SID,Location,Environment,Product,CentralServerVmId,MsiId,ManagedResourceGroupName,ManagedRgStorageAccountName,ManagedResourcesNetworkAccessType,Tag,State
+TST,eastus,NonProd,S4HANA,/subscriptions/<subscription>/resourceGroups/massreg/providers/Microsoft.Compute/virtualMachines/tstvm,/subscriptions/<subscription>/resourcegroups/JASKIRAT-RG/providers/Microsoft.ManagedIdentity/userAssignedIdentities/jaskirat-msi,mrg-TST,acssstoragetst,Public,key1 = "value1"; key2 = "value2"; key3 = "value3"; key4 = "value4",
```
Comment (kalyaninamuduri) [inline MassRegistrationOfSAPSystems/RegistrationData.csv:2]: In the example, let us please specify private as the option. Since default is public, the example having private will help

File: MassRegistrationOfSAPSystems/Readme.md [hunk 3]
```diff
@@ -16,7 +16,7 @@ The example script described in this article can be run from any Powershell 5.1
     4. SAP product for the SID being registered. Valid values are S/4HANA, ECC and Other.
     5. SAP Central Services (ASCS) virtual machine resource identifier.
     6. User assigned managed identity with the [right permissions](https://learn.microsoft.com/en-us/azure/sap/center-sap-solutions/register-existing-system#setup-user-assigned-managed-identity).
-    7. Optionally, you can provide the names for Managed Resource Group and Managed Storage Account that will be created as part of the registration process. If you do not provide these names, ACSS service deploys them using its default naming convention.
+    7. Optionally, you can provide the names for Managed Resource Group,  Managed Storage Account and Managed Resources Network Access Type that will be created as part of the registration process. If you do not provide these names, ACSS service deploys them using its default naming convention.
```
Comment (kalyaninamuduri) [inline MassRegistrationOfSAPSystems/Readme.md:19]: Let us reword this as below:

7. Optionally, you can provide the names for Managed Resource Group, Managed Storage Account and Managed Resources Network Access Type that will be created as part of the registration process. If you do not provide specific names, ACSS service deploys them using its default naming convention. 

- Managed Resources Network Access Type if unspecified will default to Public network access. Use 'private' as input value so that network access on managed storage account is restricted to specific virtual networks.
